### DISCUSSION TITLE
pyscf.scf update gto.mol object and update _vhf 2 electron integrals

### DISCUSSION THREAD

**Discussion by suarez-va** [pyscf/pyscf#2648] [Q&A] [answered]
How can I update all of my two electron integrals without having create a brand new pyscf.scf object? 

I'm implementing Ehrenfest dynamics through pyscf and I'm trying to optimize my side of the code where I'm updating a pyscf.scf object in time, both the mo_coeff and the mol object are updated. I ran into a problem where the two electron integrals would not update unless I completely reinstantiate the pyscf object to the new nuclear geometry. At the moment I'm forced to reinstantiate the pyscf.scf object every timestep which is becoming a bit of a bottleneck. Might there be a more efficient way to call the pyscf integrals outside of the pyscf.scf object?

**Comment by tberkel** [comment]
For a changing geometry of the same molecule, the simplest way is to use PySCF scanners, as described [here](https://pyscf.org/user/using.html#scanners).

**Comment by suarez-va** [comment]
Thank you for the response, these scanners seem to point in the right direction for what I want, though in my case I wouldn't want the scanner to run the scf procedure with each mole update. What I really want is to be able to call the get_fock(dm) at different nuclear geometries for a specified non-stationary density matrix.

**Comment by tberkel** [comment]
You can access AO integrals from the `mol` object alone, e.g., `mol.intor('int2e')`, which you could then contract with the DM to produce your Fock matrix. (If needed, you can also do integral transformations using the `pyscf.ao2mo` module).

**Comment by sunqm** [comment, marked as answer]
```
mf.copy().reset(mol).get_fock(mol, dm)
```
`mf.copy().reset()` will create a new instance and delete all attributes except the wavefunction information (such as `mo_coeff`, `mo_energy`)
